src/Mvc/Mvc.Core/src/ControllerBase.cs

```diff
@@ -1886,13 +1886,29 @@ public virtual ConflictObjectResult Conflict([ActionResultObjectValue] ModelStat
             string title = null,
             string type = null)
         {
-            var problemDetails = ProblemDetailsFactory.CreateProblemDetails(
-                HttpContext,
-                statusCode: statusCode ?? 500,
-                title: title,
-                type: type,
-                detail: detail,
-                instance: instance);
+            ProblemDetails problemDetails;
+            if (ProblemDetailsFactory == null)
```

Comment:
We have some precedence for having test-only code paths in Controller \ ControllerBase (e.g. https://github.com/dotnet/aspnetcore/blob/f401f9869b5c15991e6d077c4afec555509b3482/src/Mvc/Mvc.Core/src/ControllerBase.cs#L75-L78). It felt excessive to author a default implementation of the factory when it's easy to instantiate `ProblemDetails` \ `ValidationProblemDetails` in the action.